File: spring-demoapp/src/main/java/com/example/springapp/data/Director.java
package com.example.springapp.data;

import java.util.List;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.EntityListeners;
import javax.persistence.OneToMany;

@Entity
@EntityListeners(DirectorListener.class)
public class Director extends AbstractEntity {

    @Column(unique = true)
    private String name;

    @OneToMany
    private List<Movie> movies;

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public List<Movie> getMovies() {
		return movies;
	}

	public void setMovies(List<Movie> movies) {
		this.movies = movies;
	}
}

File: spring-demoapp-kotlin/src/main/java/com/example/springapp/data/Director.kt
package com.example.springapp.data

import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.OneToMany

@Entity
class Director : BaseEntity() {

    @Column(unique = true)
    var name: String? = null

    @OneToMany
    var movies: MutableList<Movie>? = null

}
